ExampleMod/Projectiles/ExampleLastPrismHoldout.cs

```diff
@@ -0,0 +1,224 @@
+﻿using ExampleMod.Items.Weapons;
+using Microsoft.Xna.Framework;
+using Microsoft.Xna.Framework.Graphics;
+using System;
+using Terraria;
+using Terraria.ID;
+using Terraria.ModLoader;
+using static Terraria.ModLoader.ModContent;
+
+namespace ExampleMod.Projectiles
+{
+	public class ExampleLastPrismHoldout : ModProjectile
+	{
+		public override string Texture => "Terraria/Projectile_" + ProjectileID.LastPrism;
+		
+		// This controls how many individual beams are fired by the Prism.
+		public const int NumBeams = 10;
+
+		// This value controls how many frames it takes for the Prism to reach "max charge". 60 frames = 1 second.
+		public const float MaxCharge = 180f;
+
+		// This value controls how many frames it takes for the beams to begin dealing damage. Before then they can't hit anything.
+		public const float DamageStart = 30f;
+
+		// This value controls how sluggish the Prism turns while being used. Vanilla Last Prism is 0.08f.
+		// Higher values make the Prism turn faster.
+		private const float AimResponsiveness = 0.08f;
+
+		// This value controls how frequently the Prism emits sound once it's firing.
+		private const int SoundInterval = 20;
+
+		// These values place caps on the mana consumption rate of the Prism.
+		// When first used, the Prism consumes mana once every MaxManaConsumptionDelay frames.
+		// Every time mana is consumed, the pace becomes one frame faster, meaning mana consumption smoothly increases.
+		// When capped out, the Prism consumes mana once every MinManaConsumptionDelay frames.
+		private const float MaxManaConsumptionDelay = 15f;
+		private const float MinManaConsumptionDelay = 5f;
+
+		// This property encloses the internal AI variable projectile.ai[0]. It makes the code easier to read.
+		private float FrameCounter {
+			get => projectile.ai[0];
+			set => projectile.ai[0] = value;
+		}
+
+		// This property encloses the internal AI variable projectile.ai[1].
+		private float NextManaFrame {
+			get => projectile.ai[1];
+			set => projectile.ai[1] = value;
+		}
+
+		// This property encloses the internal AI variable projectile.localAI[0].
+		// localAI is not automatically synced over the network, but that does not cause any problems in this case.
+		private float ManaConsumptionRate {
+			get => projectile.localAI[0];
+			set => projectile.localAI[0] = value;
+		}
+
+		public override void SetStaticDefaults()
+		{
+			DisplayName.SetDefault("Example Last Prism");
+			Main.projFrames[projectile.type] = 5;
+		}
+
+		public override void SetDefaults()
+		{
+			projectile.CloneDefaults(ProjectileID.LastPrism);
+		}
+
+		public override void AI()
```

Comment:
You should subdivide the AI method so it's easier to read/manage
e.g.
```suggestion
		public override void AI() {
			UpdateFrame();
			PlaySound();
			UpdateVisuals();
			//... etc.
		}
```